File: Python/Binary Trees/L_maxDepthBT.py
import collections


def maxDepth(root) -> int:
        if not root:
            return 0
        height = 0
        q = collections.deque([root])
        while q:
            s = len(q)
            for i in range(s):
                curr = q.popleft()
                if curr.left:
                    q.append(curr.left)
                if curr.right:
                    q.append(curr.right)
            height += 1
        return height


def maxDepth(root):
        def dfs(root,depth):
            if root is None:
                return depth
            return max(dfs(root.left, depth + 1), dfs(root.right, depth + 1))
        return dfs(root,0)

File: cpp/Binary Tree/L_maxDepthBT.cpp
#include<bits/stdc++.h>
using namespace std;

int maxDepthRec(TreeNode* root) {
        if(!root) return 0;
        int left = maxDepth(root->left);
        int right = maxDepth(root->right);
        return max(left, right) + 1;
}
int maxDepthEtr(TreeNode* root) {
        if(!root) return 0;
        int depth = 0;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()) {
            int n = q.size();
            while(n--) {
                TreeNode* curr = q.front();
                q.pop();
                if(curr->left) q.push(curr->left);
                if(curr->right) q.push(curr->right);
            }
            depth++;
        }
        return depth;
    }
int main() {

}